lte/gateway/c/oai/tasks/mme_app/mme_app_context.c

```diff
@@ -1686,96 +1684,97 @@ void mme_ue_context_update_ue_sig_connection_state(
+          LOG_MME_APP, ue_context_p->emm_context._imsi64,
+          "UE context enb_ue_s1ap_ue_id_key %ld "
+          "mme_ue_s1ap_id " MME_UE_S1AP_ID_FMT
+          ", ENB_UE_S1AP_ID_KEY could not be found",
+          ue_context_p->enb_s1ap_id_key, ue_context_p->mme_ue_s1ap_id);
     }
     ue_context_p->enb_s1ap_id_key = INVALID_ENB_UE_S1AP_ID_KEY;
 
     OAILOG_DEBUG_UE(
-      LOG_MME_APP,
-      ue_context_p->emm_context._imsi64,
-      "MME_APP: UE Connection State changed to IDLE. mme_ue_s1ap_id "
-      "= " MME_UE_S1AP_ID_FMT "\n",
-      ue_context_p->mme_ue_s1ap_id);
+        LOG_MME_APP, ue_context_p->emm_context._imsi64,
+        "MME_APP: UE Connection State changed to IDLE. mme_ue_s1ap_id "
+        "= " MME_UE_S1AP_ID_FMT "\n",
+        ue_context_p->mme_ue_s1ap_id);
 
     if (mme_config.nas_config.t3412_min > 0) {
-      // Start Mobile reachability timer only if peroidic TAU timer is not disabled
-      nas_itti_timer_arg_t timer_callback_arg = {0};
-      timer_callback_arg.nas_timer_callback =
-        mme_app_handle_mobile_reachability_timer_expiry;
-      timer_callback_arg.nas_timer_callback_arg =
-        (void *) &(ue_context_p->mme_ue_s1ap_id);
-      if (timer_setup(
-        ue_context_p->mobile_reachability_timer.sec,
-        0,
-        TASK_MME_APP,
-        INSTANCE_DEFAULT,
-        TIMER_ONE_SHOT,
-        &timer_callback_arg,
-        sizeof(timer_callback_arg),
-        &(ue_context_p->mobile_reachability_timer.id)) < 0) {
-        OAILOG_ERROR_UE(
-          LOG_MME_APP,
-          ue_context_p->emm_context._imsi64,
-          "Failed to start Mobile Reachability timer for UE id "
-          " " MME_UE_S1AP_ID_FMT "\n",
-          ue_context_p->mme_ue_s1ap_id);
-        ue_context_p->mobile_reachability_timer.id = MME_APP_TIMER_INACTIVE_ID;
+      // Start Mobile reachability timer only if peroidic TAU timer is not
+      // disabled
+      if (ue_context_p->mobile_reachability_timer.id ==
+          MME_APP_TIMER_INACTIVE_ID) {
+        // Start Mobile Reachability timer only if it is not running
+        nas_itti_timer_arg_t timer_callback_arg = {0};
+        timer_callback_arg.nas_timer_callback =
+            mme_app_handle_mobile_reachability_timer_expiry;
+        timer_callback_arg.nas_timer_callback_arg =
+            (void*) &(ue_context_p->mme_ue_s1ap_id);
+        if (timer_setup(
+                ue_context_p->mobile_reachability_timer.sec, 0, TASK_MME_APP,
+                INSTANCE_DEFAULT, TIMER_ONE_SHOT, &timer_callback_arg,
+                sizeof(timer_callback_arg),
+                &(ue_context_p->mobile_reachability_timer.id)) < 0) {
+          OAILOG_ERROR_UE(
+              LOG_MME_APP, ue_context_p->emm_context._imsi64,
+              "Failed to start Mobile Reachability timer for UE id "
+              " " MME_UE_S1AP_ID_FMT "\n",
+              ue_context_p->mme_ue_s1ap_id);
+          ue_context_p->mobile_reachability_timer.id =
+              MME_APP_TIMER_INACTIVE_ID;
```

Comment:
This is already the case due to the if condition and can be removed.

Reply:
or is this modified inside timer_setup call?

Reply:
Yes Ulas this ue_context_p->mobile_reachability_timer.id is modified in the timer_setup call. Although I don't see the chances of getting failure from this function after assignment of this timer_id because it is done almost in the end of the timer_setup call, just before adding it in the timer queue. But I left it as it is for safety purpose as it was done in the same way for other timers also.